File: src/main/java/com/sedmelluq/discord/lavaplayer/filter/FloatPcmAudioFilter.java
package com.sedmelluq.discord.lavaplayer.filter;

/**
 * Audio filter which accepts floating point PCM samples.
 */
public interface FloatPcmAudioFilter extends AudioFilter {
  /**
   * @param input An array of samples for each channel
   * @param offset Offset in the arrays to start at
   * @param length Length of the target sequence in arrays
   * @throws InterruptedException When interrupted externally (or for seek/stop).
   */
  void process(float[][] input, int offset, int length) throws InterruptedException;
}

File: main/src/main/java/lavaplayer/filter/FloatPcmAudioFilter.kt
package lavaplayer.filter

import lavaplayer.filter.AudioFilter
import kotlin.Throws
import java.lang.InterruptedException

/**
 * Audio filter which accepts floating point PCM samples.
 */
interface FloatPcmAudioFilter : AudioFilter {
    /**
     * @param input  An array of samples for each channel
     * @param offset Offset in the arrays to start at
     * @param length Length of the target sequence in arrays
     * @throws InterruptedException When interrupted externally (or for seek/stop).
     */
    @Throws(InterruptedException::class)
    fun process(input: Array<FloatArray?>, offset: Int, length: Int)
}
